src/renderer.cpp

```diff
@@ -38,11 +29,13 @@ Renderer::Renderer(Madd* context, int width, int height, const char *title):cont
 
     glEnable(GL_DEPTH_TEST);
 
-    glfwSetFramebufferSizeCallback(window, framebuffer_size_callback);
+    glfwSetFramebufferSizeCallback(window, FramebufferSizeCallback);
+}
+
+Renderer& Renderer::getInstance() {
 
-    //delete me I dare you
-    globalHeight=height;globalWidth=width;
-    rendererPointer = this;
+    static Renderer r;
+    return r;
 }
 
 Renderer::~Renderer(){
```

Comment:
Since Renderer is a singleton this destructor needs to be replaced with a Terminate Function. Otherwise we have no control over when this function runs, and since it kills the glfw context that is very bad.

Reply:
@HartleyAHartley does it need to run before/after something specific? It should run on any normal exit as it is right now.

Reply:
It needs to run after all RenderedObjects are deleted at a minimum. In my testing it works to best to call the terminator at the end of the Madd destructor.